src/ring_theory/power_series/well_known.lean

```diff
@@ -83,4 +84,38 @@ variables {A A'} (n : ℕ) (f : A →+* A')
 
 end field
 
+open ring_hom
+open finset nat
+
+variables {A : Type*} [comm_ring A]
+
+/-- Shows that `e^(aX) * e^(bX) = e^((a + b)X) ` -/
+theorem exp_mul_exp_eq_exp_add [algebra ℚ A] (a b : A) :
+  eval_mul_hom a (exp A) * eval_mul_hom b (exp A) = eval_mul_hom (a + b) (exp A) :=
+begin
+  ext, rw [coeff_mul, exp, eval_mul_hom, eval_mul_hom, eval_mul_hom],
+  simp only [coeff_mk, coe_mk, factorial], rw nat.sum_antidiagonal_eq_sum_range_succ_mk,
+  simp only [factorial, add_pow, sum_mul], apply sum_congr rfl,
+  { rintros x hx,
+    suffices : a^x * b^(n - x) * ((algebra_map ℚ A) (1 / ↑(x.factorial)) * (algebra_map ℚ A)
+     (1 / ↑((n - x).factorial))) =
+     a^x * b^(n - x) * ((↑(n.choose x) * (algebra_map ℚ A) (1 / ↑(n.factorial)))),
+    { convert this using 1; ring },
+    congr' 1,
+    rw [←map_nat_cast (algebra_map ℚ A) (n.choose x), ←map_mul, ←map_mul],
+    refine ring_hom.congr_arg _ _, rw [mul_one_div ↑(n.choose x) _, one_div_mul_one_div],
+    symmetry, rw [div_eq_iff, div_mul_eq_mul_div, one_mul, choose_eq_factorial_div_factorial],
+    norm_cast, rw cast_dvd_char_zero,
```

Comment:
@bryangingechen, did zulip decide that putting multiple tactics on the same line like this was fine?

Reply:
I think we try to avoid it unless they're really somehow accomplishing the same thing. Most of these look like they should be on separate lines.

Reply:
Changed.